woocommerce/woocommerce-gateway-stripe
# Implement Stripe Express Checkout Elements (ECE) on the Product page

Fixes #3410 

> [!IMPORTANT]
> This PR is based off the changes in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3436. This branch will need to be updated to include any future changes made to that base branch.

## Changes proposed in this Pull Request:







This PR implements Stripe's ECE (Google Pay & Apple Pay buttons) on the product page:

| Chrome | Safari |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/fa22355d-8935-48c3-9a77-e030a33939a5) | ![image](https://github.com/user-attachments/assets/af282829-fc4f-4016-b3fe-23eaf83ac6bc) |

## Testing instructions





There are **A LOT** scenarios to cover when testing these buttons from the product page. Let's start with the basics:

1. Enable the ECE feature flag in the wp-options table. 
```
update_option( '_wcstripe_feature_ece', 'yes' );
```
2. Make sure Apple Pay and Google Pay are enabled: 
![image](https://github.com/user-attachments/assets/2c3f4f0e-6c44-412a-9c9a-b5fdf630ca3a)
3. Create a simple WC Product and visit the front-facing Product page
4. Change the quantity of the product, confirm the ECE button was blocked and unblocked
5. Click on the Google Pay button
6. Confirm the order details and totals in Google Pay are correct
7. Change shipping methods between free and flat rate and confirm price is correct

| Free | Flat rate |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/824c6949-7fcd-4359-9ba0-15c7ccb98292) | ![image](https://github.com/user-attachments/assets/deda2614-d972-47fb-9745-f3be7f18b9c7)| 
8. Complete the google pay purchase
9. Confirm the order is created in WooCommerce
10. Confirm the charge in Stripe is correct
![image](https://github.com/user-attachments/assets/3accf2cd-36d9-437a-8e4f-a19c85fde47f)

---


**Post merge**
